components/dashboard/src/configurations/detail/ConfigurationDetailGeneral.tsx

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { FC } from "react";
+import { useParams } from "react-router";
+import { ConfigurationNameForm } from "./ConfigurationName";
+import { ConfigurationDetailPage } from "./ConfigurationDetailPage";
+import { useConfiguration } from "../../data/configurations/configuration-queries";
+import { RemoveConfiguration } from "./RemoveConfiguration";
+
+type PageRouteParams = {
+    id: string;
+};
+const ConfigurationDetailGeneral: FC = () => {
+    const { id } = useParams<PageRouteParams>();
+    const configurationQuery = useConfiguration(id);
+
+    return (
+        <ConfigurationDetailPage configurationQuery={configurationQuery} id={id}>
+            <ConfigurationNameForm configuration={configurationQuery.data!} />
```

Comment:
We should memoize `configurationQuery.data!` here as to not repeat ourselves.

Reply:
@selfcontained I ended up leaving it as follows. I cannot do an early return in `ConfigurationDetailGeneral.tsx`, because then not even the submenu parent would be able to handle error and loading states. Doing it in every component that requires configurations passed in to it seemed to add a lot of complexity for just getting the types to work. I was using `data!`, because the parent (`ConfigurationDetailPage`) only renders its children (requiring defined `data`) if the query is not loading, has no errors and has data.

```tsx
<ConfigurationDetailPage configurationQuery={configurationQuery} id={id}>
    {data && (
        <>
            <ConfigurationNameForm configuration={data} />
            <RemoveConfiguration configuration={data} />
        </>
    )}
</ConfigurationDetailPage>
```

Regarding the `data` variable, do you mean to call the hook 2 times? Is it acceptable to call it once and read from it subsequently?
```tsx
const configurationQuery = useConfiguration(id);
const { data } = useConfiguration(id);
```
vs
```tsx
const configurationQuery = useConfiguration(id);
const { data } = configurationQuery;
```